### ISSUE TITLE
Blocking in Chrome Incognito mode has no immediate effect to normal (non-Incognito) window ...

### ISSUE THREAD

**Issue by Alpengreis** [gorhill/uBlock#1188]
If I have to use the Chrome Incognito mode and block - for example - the iframes (3p frames) (including saving), then this has no immediate effect to the non-Incognito Window (3p frames are not blocked).  AND vice-versa ... After close and reopen Chrome completely, it's blocked too.

And - just for info: changes in both modes in the same session (not really senseful in real life, I know), changes in Incognito mode are discarded after closing Chrome completely.

PS: uBO 1.5.3 with Chrome 47.0.2526.106 m (64-bit) on Win 10 Pro x64

**Comment by IDKwhattoputhere** [comment]
Chrome's task manager says that uBO is an "anonymous" extension when in incognito. I suspect that means it's a separate instance and that's why the settings don't transfer immediately?

**Comment by gorhill** [comment]
Yes, there are two separate instances, one for normal mode, the other for incognito mode.

There is no easy way to deal with this. "Fixing" this would force one instance to reload data from storage when it changes = nasty overhead + lots of code added just to deal with this (more code = more bugs, etc.). Last time I checked, you get similar behavior with the browser itself: changing some settings won't be reflected by windows from the other mode.

In short, I currently don't plan anything for this. There is also related issue #104 -- which I haven't decided whether this really needs "fixing".

**Comment by Alpengreis** [comment]
Ok, at least I know the reason now. Thank you. I can handle this behaviour.
